_glaredb/sq-functions/parquet_scan.md

```diff
@@ -6,37 +6,163 @@ parent: SQL functions
+JSON encoded.
+
+```sql
+-- Single object.
+SELECT * FROM parquet_scan('gs://my-bucket/path/object.parquet',
+                           service_account_key => '<service_account_key>');
+-- Scan objects matching a glob.
+SELECT * FROM parquet_scan('gs://my-bucket/path/prefix_*.parquet',
+                           service_account_key => '<service_account_key>');
+-- Read all objects in a directory.
+SELECT * FROM parquet_scan('gs://my-bucket/path/*',
+                           service_account_key => '<service_account_key>');
+```
+
+Optionally, a credentials object can be created and used in place of providing
+the service account key directly.
+
+```sql
+-- Create the credentials.
+CREATE CREDENTIALS my_gcp_creds PROVIDER gcp
+	( service_account_key '<service_account_key>' );
+-- And use them in the scan.
+SELECT * FROM parquet_scan('gs://my-bucket/path/object.parquet', my_gcp_creds);
 ```
 
-| Field | Description                              |
-| ----- | ---------------------------------------- |
-| `url` | The URL or path to a parqet file to scan |
+### Objects in S3
+
+Objects in GCS can be read by providing the path to the object and credentials
+for the object, and the bucket region to `parquet_scan`.
+
+```sql
+SELECT * FROM parquet_scan('gs://my-bucket/path/object.parquet',
+                           access_key_id = '<access_key_id>',
+                           secret_access_key = '<secret_access_key>',
+                           region = 'us-east-1');
+-- Scan objects matching a glob.
+SELECT * FROM parquet_scan('gs://my-bucket/path/prefix_*.parquet',
+                           access_key_id = '<access_key_id>',
+                           secret_access_key = '<secret_access_key>',
+                           region = 'us-east-1');
+-- Read all objects in a directory.
+SELECT * FROM parquet_scan('gs://my-bucket/path/prefix_*.parquet',
+                           access_key_id = '<access_key_id>',
+                           secret_access_key = '<secret_access_key>',
+                           region = 'us-east-1');
+```
+
+Optionally, a credentials object can be created and used in place of providing
+the service account key directly.
+
+```sql
+-- Create the credentials.
+CREATE CREDENTIALS my_aws_creds PROVIDER aws
+	OPTIONS (
+        access_key_id = '<access_key_id>',
+        secret_access_key = '<secret_access_key>',
+	);
+-- And use them in the scan. Note that a region still needs to be provided
+SELECT * FROM parquet_scan('gs://my-bucket/path/object.parquet', my_gcp_creds, region => 'us-east-1');
+```
+
+### HTTP files
+
+Parquet files can also be scanned over HTTP/HTTPS.
+
+```sql
+-- Read a single file.
+select * from parquet_scan('https://my_domain.com/file.parquet');
+-- Read multiple files
+select * from parquet_scan(['https://my_domain.com/1.parquet', 'https://my_domain.com/2.parquet']);
+```
+
+Note that globbed paths are not supported when reading parquet files over HTTP/HTTPS.
```

Comment:
might make sense as a callout, it's up to you

would look like:

```
{: .warning}

> Globbed paths are not supported when reading parquet files over HTTP/HTTPS
```

or if warning feels too in-your-face, change it to `{: .important}`